geomstats/learning/riemannian_mean_shift.py

```diff
@@ -0,0 +1,177 @@
+    metric : object of class RiemannianMetric
+        Geomstats Riemannian metric associated to the space used.
+    bandwidth : float
+        Size of neighbourhood around each center. All points in 'bandwidth'
+        size around center are considered for calculating new mean centers.
+    tol : float
+        Stopping condition. Computation of subsequent mean centers is stopped
+        when the distance between them is less than 'tol'.
+        Optional, default : 1e-2.
+    n_centers : int
+        Number of centers.
+        Optional, default : 1.
+    n_jobs : int
+        Number of parallel threads to be initiated for parallel jobs.
+        Optional, default : 1.
+    max_iter : int
+        Upper bound on total number of iterations for the centers to converge.
+        Optional, default : 100.
+    init_centers : str
+        Initializing centers, either from the given input points or
+        random points uniformly distributed in the input manifold.
+        Optional, default : "from_points".
+    kernel : str
+        Weighing function to assign kernel weights to each center.
+        Optional, default : "flat".
+    """
+
+    def __init__(
+        self,
+        manifold,
+        metric,
+        bandwidth,
+        tol=1e-2,
+        n_centers=1,
+        n_jobs=1,
+        max_iter=100,
+        init_centers="from_points",
+        kernel="flat",
+        **frechet_mean_kwargs
+    ):
+        self.manifold = manifold
+        self.metric = metric
+        self.bandwidth = bandwidth
+        self.tol = tol
+        self.mean = FrechetMean(self.metric, **frechet_mean_kwargs)
+        self.n_centers = n_centers
+        self.n_jobs = n_jobs
+        self.max_iter = max_iter
+        self.init_centers = init_centers
+        self.kernel = kernel
+        self.centers = None
+
+    def dist_intersets(self, points_a, points_b, **joblib_kwargs):
+        """Parallel computation of distances between two sets of points."""
+        n_a, n_b = points_a.shape[0], points_b.shape[0]
+
+        @joblib.delayed
+        @joblib.wrap_non_picklable_objects
+        def pickable_dist(x, y):
+            """Riemannian metric between points x & y"""
+            return self.metric.dist(x, y)
+
+        pool = joblib.Parallel(n_jobs=self.n_jobs, **joblib_kwargs)
+        out = pool(
+            pickable_dist(point_a, point_b)
+            for point_a in points_a
+            for point_b in points_b
+        )
+
+        return gs.array(out).reshape((n_a, n_b))
+
+    def fit(self, x, y=None):
+        """
+        Fit centers in all the input points to find
+        'n_centers' number of clusters.
+        """
+
+        @joblib.delayed
+        @joblib.wrap_non_picklable_objects
+        def pickable_mean(points, weights):
+            """Frechet Mean of all points weighted by weights.
+
+            Parameters
+            ----------
+            points : array-like,
+                     Clusters of points.
```

Comment:
Docstring conventions:
- Indentation should be +4 for parameter's description
- Include the shape of the array after the `,` (see examples of docstrings in geomstats)

Reply:
Incorrect indentation, see above